wrappers/core18_test.go

```diff
@@ -227,6 +229,39 @@ func (s *servicesTestSuite) TestAddSnapServicesForSnapdOnClassic(c *C) {
 	c.Check(s.sysdLog, IsNil)
 }
 
+func (s *servicesTestSuite) TestAddSessionServicesWithReadOnlyFilesystem(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	restoreEnsureDirState := wrappers.MockEnsureDirState(func(dir string, glob string, content map[string]osutil.FileState) (changed, removed []string, err error) {
+		return nil, nil, &os.PathError{Err: syscall.EROFS}
```

Comment:
Is this a realistic mock? Is there any chance that we are returning a naked EROFS from the real code?

Reply:
Yes, I it reflects error wrapping that I saw when provoking this scenario with the spread test.  But I like your idea below.